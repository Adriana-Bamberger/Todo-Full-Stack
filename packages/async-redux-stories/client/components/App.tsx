import ErrorMessage from './ErrorMessage.tsx'
import LoadSubreddit from './LoadSubreddit.tsx'
import SubredditList from './SubredditList.tsx'
import WaitIndicator from './WaitIndicator.tsx'

function App() {
  return (
    <div className="app">
      <ErrorMessage />
      <LoadSubreddit>
        <WaitIndicator />
      </LoadSubreddit>
      <SubredditList />
    </div>
  )
}

export default App
