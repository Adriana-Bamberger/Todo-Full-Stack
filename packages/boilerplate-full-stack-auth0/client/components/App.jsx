import Nav from './Nav'
import PingRoutes from './PingRoutes'
import Registration from './Registration'
import Users from './Users'
import { Routes, Route } from 'react-router-dom'
import { useAuthCache } from '../auth0-utils'

function App() {
  useAuthCache()

  return (
    <div className="app">
      <Routes>
        <Route path="/" element={<Nav />} />
        <Route path="/" element={<Users />} />
        <Route path="/" element={<PingRoutes />} />
        <Route path="/profile" element={<Registration />} />
      </Routes>
    </div>
  )
}

export default App
