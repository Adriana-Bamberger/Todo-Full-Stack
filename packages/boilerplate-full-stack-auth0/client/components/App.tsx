import Nav from './Nav.tsx'
import PingRoutes from './PingRoutes.tsx'
import Registration from './Registration.tsx'
import Users from './Users.tsx'
import { Routes, Route } from 'react-router-dom'
import { useAuthCache } from '../auth0-utils.ts'

function App() {
  useAuthCache()

  return (
    <div className="app">
      <Routes>
        <Route path="/" element={<Nav />} />
        <Route path="/" element={<Users />} />
        <Route path="/" element={<PingRoutes />} />
        <Route path="/profile" element={<Registration />} />
      </Routes>
    </div>
  )
}

export default App
