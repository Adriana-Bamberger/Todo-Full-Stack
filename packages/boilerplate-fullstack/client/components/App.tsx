import { useEffect } from 'react'
import { useAppDispatch, useAppSelector } from '../hooks.ts'

import { fetchFruits } from '../actions/index.ts'

function App() {
  const fruits = useAppSelector((state) => state.fruits)
  const dispatch = useAppDispatch()
  useEffect(() => {
    dispatch(fetchFruits())
  }, [dispatch])

  return (
    <>
      <div className="app">
        <h1>Fullstack Boilerplate - with Fruits!</h1>
        <ul>
          {fruits.map((fruit) => (
            <li key={fruit}>{fruit}</li>
          ))}
        </ul>
      </div>
    </>
  )
}

export default App
