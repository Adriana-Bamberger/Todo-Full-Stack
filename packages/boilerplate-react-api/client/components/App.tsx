import { useState, useEffect } from 'react'
import { getGreeting } from '../apiClient.ts'
import { useQuery } from '@tanstack/react-query'

const App = () => {
  const [count, setCount] = useState(0)

  const {
    data: greeting,
    isError,
    isLoading,
  } = useQuery({ queryKey: ['greeting', count], queryFn: getGreeting })

  if (isLoading) return <p>Loading...</p>

  return (
    <>
      {count}
      <h1>{greeting}</h1>
      {isError && (
        <p style={{ color: 'red' }}>
          There was an error retrieving the greeting.
        </p>
      )}
      <button onClick={() => setCount(count + 1)}>Click</button>
    </>
  )
}

export default App
