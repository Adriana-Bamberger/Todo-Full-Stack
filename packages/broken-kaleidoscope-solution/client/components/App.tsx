import Pixel from './Pixel'

function App() {
  return (
    <>
      {Array.from({ length: 100 }).map((_, i) => (
        <Pixel key={i} />
      ))}
    </>
  )
}

export default App
