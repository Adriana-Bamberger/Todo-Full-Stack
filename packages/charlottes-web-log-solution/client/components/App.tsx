import Header from './Header'
import headerProps from '../data/header'

import Footer from './Footer'
import footerProps from '../data/footer'

import Posts from './Posts'
import OtherBlogs from './OtherBlogs'
import RecentEntries from './RecentEntries'

function App() {
  return (
    <div className="app">
      <Header {...headerProps} />
      <div className="body-container">
        <OtherBlogs />
        <Posts />
        <RecentEntries />
      </div>
      <Footer {...footerProps} />
    </div>
  )
}

export default App
