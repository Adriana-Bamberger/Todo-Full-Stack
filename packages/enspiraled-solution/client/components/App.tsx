import { useCallback, useState } from 'react'
import Circle from './Circle'

interface Props {
  width: number
  height: number
}

function App({ width, height }: Props) {
  const [token, setToken] = useState(1)

  // STRETCH: Add a button to the page that will clear the screen and start over
  // TODO: I don't think this is a nice way to do this
  const handleReset = useCallback(() => {
    setToken(Math.random())
  }, [])

  const circle = {
    cx: width / 2,
    cy: height / 2,
    level: 0,
    r: 256,
  }

  return (
    <>
      <button onClick={handleReset}>Reset</button>
      <svg width={width} height={height}>
        <Circle {...circle} key={token} />
      </svg>
    </>
  )
}

export default App
