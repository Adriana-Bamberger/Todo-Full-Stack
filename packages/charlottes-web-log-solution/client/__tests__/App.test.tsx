// @vitest-environment jsdom
import { describe, it, expect, afterEach } from 'vitest'
import { screen, render, cleanup } from '@testing-library/react'
import matchers from '@testing-library/jest-dom/matchers'

expect.extend(matchers)
afterEach(cleanup)

import App from '../components/App'

describe('<App />', () => {
  it('looks like this', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  it("Links to wilber's blog", () => {
    render(<App />)

    const link = screen.getByRole('link', {
      name: 'Wilber',
    }) as HTMLAnchorElement
    expect(link.href).toBe('https://wilber.net/blog')
  })
})
