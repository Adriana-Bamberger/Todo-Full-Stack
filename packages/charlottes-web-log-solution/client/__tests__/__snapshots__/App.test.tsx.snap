// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App /> > looks like this 1`] = `
<div>
  <div
    class="app"
  >
    <div
      class="header"
    >
      <h1>
        <a
          href="/"
        >
          Charlotte's Web Log
        </a>
      </h1>
    </div>
    <div
      class="body-container"
    >
      <div
        class="other-blogs"
      >
        <header>
          Other Blogs
        </header>
        <ul>
          <li>
            <a
              href="https://wilber.net/blog"
            >
              Wilber
            </a>
          </li>
          <li>
            <a
              href="https://templeton.net/blog"
            >
              Templeton
            </a>
          </li>
          <li>
            <a
              href="https://john-arable.net/blog"
            >
              John Arable
            </a>
          </li>
          <li>
            <a
              href="https://fern-arable.net/blog"
            >
              Fern Arable
            </a>
          </li>
          <li>
            <a
              href="https://homer-zuckerman.net/blog"
            >
              Homer Zuckerman
            </a>
          </li>
          <li>
            <a
              href="https://old-sheep.net/blog"
            >
              Old Sheep
            </a>
          </li>
          <li>
            <a
              href="https://dr-dorian.net/blog"
            >
              Dr. Dorian
            </a>
          </li>
          <li>
            <a
              href="https://uncle.net/blog"
            >
              Uncle
            </a>
          </li>
        </ul>
      </div>
      <div
        class="posts"
      >
        <div
          class="post"
        >
          <h2>
            The beginning of an idea
          </h2>
          <div
            class="date"
          >
            15 July 1953
          </div>
          <p>
            Day after day the spider waited, head­down, for an idea to come to her. Hour by hour she sat motionless, deep in thought. Having promised Wilbur that she would save his life, she was determined to keep her promise. Charlotte was naturally patient. She knew from experience that if she waited long enough, a fly would come to her web; and she felt sure that if she thought long enough about Wilbur's problem, an idea would come to her mind. Finally, one morning toward the middle of July, the idea came. "Why, how perfectly simple!" she said to herself. "The way to save Wilbur's life  is  to  play a trick on Zuckerman. If I can fool a bug," thought Charlotte, "I can surely fool a man. People are not as smart as bugs."
          </p>
          <p>
            Wilbur walked into his yard just at that moment.
          </p>
          <p>
            "What are you thinking about, Charlotte?" he asked.
          </p>
          <p>
            "I was just thinking," said the spider, "that people are very gullible."
          </p>
          <p>
            "What does 'gullible' mean?"
          </p>
          <p>
            "Easy to fool," said Charlotte.
          </p>
          <p>
            "That's a mercy," replied Wilbur, and he lay down in the shade of his fence and went fast asleep. The spider, however, stayed wide awake, gazing affectionately at him and making plans for his future. Summer was half gone. She knew she didn't have much time.
          </p>
          <div
            class="comment-count"
          >
            4
             comments
          </div>
        </div>
        <div
          class="post"
        >
          <h2>
            Bad News
          </h2>
          <div
            class="date"
          >
            24 June 1953
          </div>
          <p>
            Wilbur liked Charlotte better and better each day. Her campaign against insects seemed sensible and useful. Hardly anybody around the farm had a good word to say for a fly. Flies spent their time pestering others. The cows hated them. The horses detested them. The sheep loathed them. Mr. and Mrs. Zuckerman were always complaining about them, and putting up screens.
          </p>
          <p>
            Wilbur admired the way Charlotte managed. He was particularly glad that she always put her victim to sleep before eating it.
          </p>
          <p>
            "It's real thoughtful of you to do that, Charlotte," he said.
          </p>
          <p>
            "Yes," she replied in her sweet, musical voice, "I always give them an anaesthetic so they won't feel pain. It's a little service I throw in."
          </p>
          <p>
            As the days went by, Wilbur grew and grew. He ate three big meals a day. He spent long hours lying on his side, half asleep, dreaming pleasant dreams. He enjoyed good health and he gained a lot of weight. One afternoon, when Feen was sitting on her stool, the oldest sheep walked into the barn, and stopped to pay a call on Wilbur.
          </p>
          <p>
            "Hello!" she said. "Seems to me you're putting on weight."
          </p>
          <p>
            "Yes, I guess I am," replied Wilbur. "At my age it's a good idea to keep gaining."
          </p>
          <p>
            "Just the same, I don't envy you," said the old sheep. "You know why they're fattening you up, don't you?"
          </p>
          <p>
            "No," said Wilbur.
          </p>
          <p>
            "Well, I don't like to spread bad news," said the sheep, "but they're fattening you up because they're going to kill you, that's why."
          </p>
          <div
            class="comment-count"
          >
            8
             comments
          </div>
        </div>
        <div
          class="post"
        >
          <h2>
            Summer Days
          </h2>
          <div
            class="date"
          >
            10 June 1953
          </div>
          <p>
            The early summer days on a farm are the happiest and fairest days of the year. Lilacs bloom and make the air sweet, and then fade. Apple blossoms come with the lilacs, and the bees visit around among the apple trees. The days grow warm and soft. School ends, and children have time to play and to fish for trouts in the brook. Avery often brought a trout home in his pocket, warm and stiff and ready to be fried for supper.
          </p>
          <p>
            Now that school was over, Fern visited the barn almost every day, to sit quietly on her stool. The animals treated her as an equal. The sheep lay calmly at her feet.
          </p>
          <p>
            Around the first of July, the work horses were hitched to the mowing machine, and Mr. Zuckerman climbed into the seat and drove into the field. All morning you could hear the rattle of the machine as it went round and round,while the tall grass fell down behind the cutter bar in long green swathes. Next day, if there was no thunder shower, all hands would help rake and pitch and load, and the hay would be hauled to the Summer Days in the high hay wagon, with Fern and Avery riding at the top  of the load. Then the hay would be hoisted, sweet and warm, into the big loft, until the whole bam seemed like a wonderful bed of timothy and clover. It was fine to jump in, and perfect to hide in. And sometimes A very would find a little grass snake in the hay, and would add it to the other things in his pocket.
          </p>
          <p>
            Early summer days are a jubilee time for birds. In the fields, around the house, in the barn, in the woods, in the swamp - everywhere love and songs and nests and eggs. From the edge of the woods, the white­throated sparrow (which must come all the way from Boston) calls, "Oh, Peabody, Peabody, Peabody!" On an apple bough, the phoebe teeters and wags its tail and says, "Phoebe, phoe-bee!" The song sparrow, who knows how brief and lovely life is, says, "Sweet, sweet, sweet interlude; sweet, sweet, sweet interlude." If you enter the barn, the swallows swoop down from their nests and scold. "Cheeky, cheeky!" they say.
          </p>
          <div
            class="comment-count"
          >
            9
             comments
          </div>
        </div>
        <div
          class="post"
        >
          <h2>
            My first encounter with Wilber
          </h2>
          <div
            class="date"
          >
            15 October 1952
          </div>
          <p>
            "My name is Charlotte."
          </p>
          <p>
            "Charlotte what?" asked Wilbur, eagerly.
          </p>
          <p>
            "Charlotte A. Cavatica. But just call me Charlotte."
          </p>
          <p>
            "I think you're beautiful," said Wilbur.
          </p>
          <p>
            "Well, I am pretty," replied Charlotte. "There's no denying that. Almost all spiders are rather nice-looking. I'm not as flashy as some, but I'll do. I wish I could see you, Wilbur, as clearly as you can see me."
          </p>
          <p>
            "Why can't you?" asked the pig. "I'm right here."
          </p>
          <p>
            "Yes, but I'm near-sighted," replied Charlotte. "I've always been dreadfully near-sighted. It's good in some ways, not so good in others. Watch me wrap up this fly."
          </p>
          <p>
            A fly that had been crawling along Wilbur's trough had flown up and blunde red into the lower part of Charlotte's web and was tangled in the sticky threads. The fly was beating its wings furiously, trying to break loose and free itself.
          </p>
          <p>
            "First," said Charlotte, "I dive at him." She plunged Charlotte's Web headfirst toward the fly. As she dropped, a tiny silken thread unwound from her rear end.
          </p>
          <p>
            "Next, I wrap him up." She grabbed the fly, threw a few jets of silk around it, and rolled it over and over, wrapping it so that it couldn't move. Wilbur watched in horror. He could hardly believe what he was seeing, and although he detested flies, he was sorry for this one.
          </p>
          <p>
            "There!" said Charlotte. "Now I knock him out, so he'll be more comfor table." She bit the fly. "He can't feel a thing now," she remarked. "He'll make a perfect breakfast for me."
          </p>
          <div
            class="comment-count"
          >
            5
             comments
          </div>
        </div>
      </div>
      <div
        class="recent-entries"
      >
        <header>
          Recent Entries
        </header>
        <ul>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/a-warm-wind"
            >
              A warm wind
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/last-day"
            >
              Last day
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/the-hour-of-triumph"
            >
              The hour of triumph
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/the-egg-sac"
            >
              The egg sac
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/the-cool-of-the-evening"
            >
              The cool of the evening
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/uncle"
            >
              Uncle
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/off-to-the-fair"
            >
              Off to the fair
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/the-crickets"
            >
              The crickets
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/dr-dorian"
            >
              Dr. Dorian
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/good-progress"
            >
              Good progress
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/a-meeting"
            >
              A meeting
            </a>
          </li>
          <li
            class="recent-entry"
          >
            <a
              href="https://charlotte.net/blog/the-miracle"
            >
              The miracle
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="footer"
    >
      <p>
        1980
      </p>
      <p>
        E. B. White
      </p>
    </div>
  </div>
</div>
`;
